src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -187,6 +187,21 @@ object JoinIndexRule
   private def isPlanLinear(plan: LogicalPlan): Boolean =
     plan.children.length <= 1 && plan.children.forall(isPlanLinear)
 
+  /**
+   * Check if the candidate plan is already modified by Hyperspace or not.
+   * This can be determined by a specific key in options of HadoopFsRelation.
+   *
+   * @param plan Logical plan.
+   * @return true if the relation in the plan is modified by Hyperspace.
+   */
+  private def isPlanModified(plan: LogicalPlan): Boolean = {
+    plan.collect {
+      case LogicalRelation(HadoopFsRelation(_, _, _, _, _, options), _, _, _)
+          if (options.contains(IndexConstants.INDEX_RELATION_IDENTIFIER_KEY)) =>
```

Comment:
Can we move this functionality to `RuleUtils` so it can be reused across rules? Something like:
```scala
def isIndexApplied(relation: HadoopFsRelation): Boolean = {
  ...  
}
```